mbedtls/src/ssl/context.rs

```diff
@@ -284,15 +331,29 @@ impl<T> Context<T> {
     pub fn config(&self) -> &Arc<Config> {
         &self.config
     }
-    
+
+    pub(super) fn close_notify(&mut self) -> Result<()> {
+        unsafe {
+            ssl_close_notify(self.into()).into_result().map(|_| ())
+        }
+    }
+
     pub fn close(&mut self) {
+        let _ = self.close_notify();
+        self.drop_io();
+    }
+
+    pub(super) fn clear_bio(&mut self) {
```

Comment:
Mark this function as ``unsafe`` if the whole body is ``unsafe``; add SAFETY comment here as well.

Reply:
Nice catch!

Reply:
If it’s mostly unsafe because of FFI it could be better the other way, it depends.

Reply:
I don't think this is unsafe.

Reply:
Same idea, I have add comments about why it's safe

Reply:
Should be a regular comment, not a doc comment.

Reply:
Should be a regular comment, not a doc comment. Safety information is normally put on the `unsafe` block and not on the containing function.

Reply:
Got it

Reply:
fix it